include/rtc.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * @file rtc.h
+ * @brief Real-Time Clock Subsystem
+ * @ingroup rtc
+ */
+#ifndef __LIBDRAGON_RTC_H
+#define __LIBDRAGON_RTC_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @addtogroup rtc
+ * @{
+ */
+
+/**
+ * @brief Structure for storing RTC time data.
+ */
+typedef struct rtc_time_t
+{
+    /** @brief Year. [1900-20XX] */
+    uint16_t year;
+    /** @brief Month. [0-11] */
+    uint8_t month;
+    /** @brief Day of month. [1-31] */
+    uint8_t day;
+    /** @brief Hours. [0-23] */
+    uint8_t hour;
+    /** @brief Minutes. [0-59] */
+    uint8_t min;
+    /** @brief Seconds. [0-59] */
+    uint8_t sec;
+    /** @brief Day of week. [0-6] (Sun-Sat) */
+    uint8_t week_day;
+} rtc_time_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+bool joybus_rtc_is_present( void );
```

Comment:
Do we really need the joybus prefix in all these names?

Reply:
We do if we ever want to support the N64DD RTC, which has a completely different interface.

Reply:
I see. OTOH I'm not sure if the average libdragon user is interested in which RTC there is, rather than "a RTC", so maybe the n64dd rtc should be implemented behind the same API?

Reply:
I have removed the joybus prefix from the API; most of the static methods still have the joybus_ prefix if they are actually joybus-specific. If/when we add support for the N64DD RTC, it can use the same interface and the LibDragon RTC subsystem can abstract away the differences